lib/multi.c

```diff
@@ -1357,6 +1364,11 @@ static CURLMcode multi_wait(struct Curl_multi *multi,
       return CURLM_INTERNAL_ERROR;
     }
 #endif
+    if(nfds >= ufds_len) {
+      if(ufds_resize(&ufds, &ufds_len, 100, &ufds_malloc))
+        return CURLM_OUT_OF_MEMORY;
+    }
+    DEBUGASSERT(nfds < ufds_len);
```

Comment:
if the size could be larger then why not just use it instead
```suggestion
    if(nfds >= ufds_len) {
      size_t needed = nfds - ufds_len;
      if(ufds_resize(&ufds, &ufds_len, needed, &ufds_malloc))
        return CURLM_OUT_OF_MEMORY;
    }
```

Reply:
also I would call it ufds_increase instead of resize because you're passing a number to increase, either that or leave it named resize and don't pass the number to increase, pass the new size instead

ufds_resize(&ufds, &ufds_len, nfds +1, &ufds_malloc)

Reply:
Will  change the name as suggested.